versions/1.1/SPEC.md

```diff
@@ -0,0 +1,5846 @@
+
+**Parameters**
+
+1. `Array[Pair[P, Y]]`: Array of `Pair`s to convert to a `Map`.
+
+**Returns**: `Map[P, Y]` of the elements in the input array.
+
+**Example**
+
+```wdl
+Array[Pair[String, Int]] x = [("a", 1), ("c", 3), ("b", 2)]
+Boolean is_true = as_map(x) == {"a": 1, "c": 3, "b": 2}
+
+Array[Pair[String, Pair[File,File]]] y = [("a", ("a.bam", "a.bai")), ("b", ("b.bam", "b.bai"))
+Boolean is_true2 = as_map(y) == {"a": ("a.bam", "a.bai"), "b": ("b.bam", "b.bai")}
+
+# this fails with an error - the "a" key is duplicated
+Boolean bad = as_map([("a", 1), ("a", 2)])  
+```
+
+## ✨ Array[P] keys(Map[P, Y])
+
+Creates an `Array` of the keys from the input `Map`, in the same order as the elements in the map.
+
+**Parameters**
+
+1. `Map[P, Y]`: `Map` from which to extract keys.
+
+**Returns**: `Array[P]` of the input `Map`s keys.
+
+**Example**
+
+```wdl
+workflow foo {
+  Map[String,Int] x = {"a": 1, "b": 2, "c": 3}
+  Boolean is_true = keys(x) == ["a", "b", "c"]
+
+  Map[String, Pair[File, File]] str_to_files = {"a": ("a.bam", "a.bai"), "b": ("b.bam", "b.bai")}
+  scatter (item in str_to_files) {
+    String key = item.left
+  }
+  Array[String] str_to_files_keys = key
+  Boolean is_truew = str_to_files_keys == keys(str_to_files)
+}
+```
+
+## ✨ Map[P, Array[Y]] collect_by_key(Array[Pair[P, Y]])
+
+Given an `Array` of `Pair`s, creates a `Map` in which the right elements of the `Pair`s are grouped by the left elements. In other words, the input `Array` may have multiple `Pair`s with the same (primitive) key - rather than causing an error (as would happen with [`as_map`](#-mapp-y-as_maparraypairp-y)), all the values with the same key are grouped together into an `Array`.
+
+The order of the keys in the output `Map` is the same as the order of their first occurrence in the input `Array`. The order of the elements in the `Map` values is the same as their order of occurrence in the input `Array`.
+
+**Parameters**
+
+1. `Array[Pair[P, Y]]`: `Array` of `Pair`s to group.
+
+**Returns**: `Map` of keys to `Array`s of values.
+
+**Example**
+
+```wdl
+Array[Pair[String, Int]] x = [("a", 1), ("b", 2), ("a", 3)]
+Boolean is_true = as_map(x) == {"a": [1, 3], "b": [2]}
+
+Array[Pair[String, Pair[File, File]]] y = [
+  ("a", ("a_1.bam", "a_1.bai")), ("b", ("b.bam", "b.bai")), ("a", ("a_2.bam", "a_2.bai"))
+]
+Boolean is_true2 = as_map(y) == {
+  "a": [("a_1.bam", "a_1.bai"), ("a_2.bam", "a_2.bai")], 
+  "b": [("b.bam", "b.bai")]
+}
+```
+
+## Boolean defined(X?)
+
+Tests whether the given optional value is defined, i.e. has a non-`None` value.
+
+**Parameters**
+
+1. `X?`: optional value of any type.
+
+**Returns**: `false` if the input value is `None`, otherwise `true`.
+
+**Example**
+
+```wdl
+task say_hello {
+  input {
+    String name
+  }
+  command { echo ~{name} }
+}
+
+workflow wf {
+  input {
+    String? s
+  }
+  if (defined(s)) {
+    call say_hello { input: name = select_first([s]) }
+  }
+}
+```
+
+## X select_first(Array[X?]+)
```

Comment:
Instead of using X here as the return type (since you usually use it as part of the variable itself), consider using `R` as you mention at the top of the standard library section:
"The signature of each function is given as `R func_name(T1, T2, ...)`, where `R` is the return type "

Replace with: ## R select_first(Array[X?]+)

Reply:
No, X is a generic type. It means that the return type is the same as the Array element type.